icu4j/main/core/src/main/java/com/ibm/icu/util/IslamicCalendar.java

```diff
@@ -826,6 +911,88 @@ protected int handleGetExtendedYear() {
         return year;
     }
 
+    /**
+     * Internal function shared by handleComputeFieldsX functions for
+     * different calendars.
+     */
+    private void internalSetYMD(int year, int month, long days) {
+        int dayOfMonth = (int)(days - monthStart(year, month)) + 1;
+        // Now figure out the day of the year.
+        int dayOfYear = (int)(days - monthStart(year, 0) + 1);
+        internalSet(ERA, 0);
+        internalSet(YEAR, year);
+        internalSet(EXTENDED_YEAR, year);
+        internalSet(MONTH, month);
+        internalSet(ORDINAL_MONTH, month);
+        internalSet(DAY_OF_MONTH, dayOfMonth);
+        internalSet(DAY_OF_YEAR, dayOfYear);
+    }
+
+    /**
+     * handleComputeFields function for Islamic Civil calendar.
+     */
+    private void handleComputeFieldsIslamicCivil(long days) {
+        // Use the civil calendar approximation, which is just arithmetic
+        int year  = (int)Math.floor( (30 * days + 10646) / 10631.0 );
```

Comment:
What are the magic numbers here?

Reply:
The same magic number as my mechanical refactoring, my guess is not better than yours. 
I do not want to change the outcome. Just want to reduce run time if (cType == ) or switch (cType) ... check here  after the refactoring

Reply:
Fair enough...